core/dive.c

```diff
@@ -3570,6 +3582,22 @@ int split_dive(struct dive *dive)
 	return 0;
 }
 
+void split_dive_at_time(struct dive *dive, int time)
+{
+	int i = 0;
+	struct sample *sample = dive->dc.sample;
+
+	if (!dive)
+		return;
+	while(sample->time.seconds < time) {
+		++sample;
+		i++;
+		if (dive->dc.samples == i)
+			return;
+	}
```

Comment:
Any reason to check for `dive->dc.samples==i` but not `i==0`?
Given the call in the next line, `i <= 1` may be even better.

Reply:
I'd agree. Splitting at sample #0 or #1 simply makes no sense

Reply:
Also note that by clicking to the left of the actual profile one can generate timestamps *before* the dive started.

Reply:
Only checking against dc.samples to make sure loop does not run off the end of the sample array. I pushed an updated version that checks the split point is well into the dive (from both ends) in split_dive_at (which has sample numbers).